src/porepy/numerics/vem/vem_source.py

```diff
@@ -66,19 +66,28 @@ def __init__(self, dim_max, physics='flow'):
         self.discr = Integral(self.physics)
         self.coupling_conditions = None
 
-        kwargs = {"discr_ndof": self.discr.ndof,
+        kwargs = {"discr_ndof": self.__ndof__,
                   "discr_fct": self.__matrix_rhs__}
         self.solver = Coupler(coupling = None, **kwargs)
         SolverMixDim.__init__(self)
 
+    def __ndof__(self, g):
+        # The highest dimensional problem has the standard number of dof
```

Comment:
Why would this be different in different dimensions? I don't see it, so at least a comment is due.

Reply:
it's again the fact of dfn source term. in the intersection we have only num_cells dof while in the fractures num_cells+num_faces. it's because the equations are not uniforms on each dimension

Reply:
it's the same concept of __matrix__rhs